### DISCUSSION TITLE
No sound after instlling pulseaudio module on Debin 11.7.0

### DISCUSSION THREAD

**Discussion by beppo-ivel** [neutrinolabs/pulseaudio-module-xrdp#97] [Q&A] [answered]
Before building and installing xrdp-pulseaudio-module on Debian 11.7.0 I checked my setup:

```
$ dpkg-query -l '*xrdp*'
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                      Version      Architecture Description
+++-=========================-============-============-===============================================
ii  xorgxrdp                  1:0.2.12-1   amd64        Remote Desktop Protocol (RDP) modules for X.org
ii  xrdp                      0.9.12-1.1   amd64        Remote Desktop Protocol (RDP) server
un  xrdp-pulseaudio-installer <none>       <none>       (no description available)
```

I run `pavumeter`  and playback a YouTube video in Firefox. I can see volume meter moving but I don't hear any sound:

![pavumeter](https://github.com/neutrinolabs/pulseaudio-module-xrdp/assets/72635809/902b73a7-072d-4097-9e14-0e1cdb17baf5)

If I run `$ sudo aplay /usr/share/sounds/alsa/Front_Center.wav` I can hear the sound on remote machine.

After this checkup I followed [build](https://github.com/neutrinolabs/pulseaudio-module-xrdp/wiki/Build-on-Debian-or-Ubuntu) and install instructions:

```
$ ls $(pkg-config --variable=modlibexecdir libpulse) | grep xrdp
module-xrdp-sink.la
module-xrdp-sink.so
module-xrdp-source.la
module-xrdp-source.so

```

But behavior is same as before I installed pulsaudio module. `sudo aplay` plays back a sound but if I playback sound under my user there is no sound. And pavumeter looks same as in the screenshot above. Does that mean there is no signal level of xrdp sink? How to fix that problem?

**Comment by matt335672** [comment]
Is the `load_pa_modules.sh` script in `/etc/xdg/autostart`?

What happens if you run it manually?

**Reply by beppo-ivel** [reply]
@matt335672  I ran the script manually but nothing changed, same error picture as described above.

```
$ cat /etc/xdg/autostart/pulseaudio-xrdp.desktop 
[Desktop Entry]
Version=1.0
Name=PulseAudio xrdp modules
Comment=Load PulseAudio Modules for xrdp
Exec=/usr/libexec/pulseaudio-module-xrdp/load_pa_modules.sh
Terminal=false
Type=Application
Categories=
GenericName=

$ /usr/libexec/pulseaudio-module-xrdp/load_pa_modules.sh 
14
- pulseaudio xrdp-sink loaded
- pulseaudio xrdp-sink set as default
15
- pulseaudio xrdp-source loaded
- pulseaudio xrdp-source set as default

```

**Comment by matt335672** [comment]
That all looks fine to me.

Suggest you trace back to your client sound installation.

Have you enabled the RDP sound channels on your RDP client? For example on the Windows 10 client, if you're playing back only I'd recommend these settings:-

![image](https://github.com/neutrinolabs/pulseaudio-module-xrdp/assets/30179339/d1648e70-29d7-46c7-b2d3-e9fac7689b26)

Also, check sounds are working as expected on your client.

**Reply by beppo-ivel** [reply]
> Have you enabled the RDP sound channels on your RDP client?

The sound should be played on remote computer, I just want control it remotely. And `sudo aplay /usr/share/sounds/alsa/Front_Center.wav` is working as expected.

**Reply by matt335672** [reply]
Do I understand this right?

You're logging in on a Windows 10 PC to a Debian 11 PC. You want to play the sound on speakers connected to the Debian 11 PC and not to the Windows 10 PC(?)

**Reply by beppo-ivel** [reply]
correct,  sitting in front of win10 pro version 22H2 and  connected to Debian 11.7.0 remotely using  mstsc.exe. And I want to play sound over speakers which are connected to Debian PC.

**Reply by matt335672** [reply, marked as answer]
That's not a standard xrdp configuration.

The pulseaudio modules are designed to carry sound over the connection so that the speakers and microphone of the windows PC are used. If you're not using that functionality, you don't need these modules.

I suggest you remove the pulseaudio modules, `/etc/xdg/autostart/pulseaudio-xrdp.desktop` and `/usr/libexec/pulseaudio-module-xrdp/load_pa_modules.sh`. That might be sufficient. If not, I tink you'll  need to load the modules your sound card uses into the user instance of pulsaudio. That, however, is not an xrdp problem.
